segmentation/utils/synthetic_spatial_datagen.py

```diff
@@ -222,10 +219,142 @@ def generate_test_label_map(size_img=(1024, 1024), num_A=100, num_B=100, num_C=1
 
     # now generate the sample xarray
     sample_img = np.zeros((1, size_img[0], size_img[1], 1)).astype(np.int16)
-    sample_img[0, :, :, 0] = copy.deepcopy(label_mat)
+    sample_img[0, :, :, 0] = deepcopy(label_mat)
     sample_img_xr = xr.DataArray(sample_img,
                                  coords=[[1], range(size_img[0]), range(size_img[1]), ['segmentation_label']],
                                  dims=['fovs', 'rows', 'cols', 'channels'])
 
     # and return the xarray to pass into calc_dist_matrix, plus the centroid_indices to readjust it
     return sample_img_xr, centroid_indices
+
+
+def generate_two_cell_test_signal_data(size_img, radius, expression,
+                                       pattern, cell_region, cell_center):
+    """
+    This function generates the signal-level test data for a specified channel (nuclear and membrane)
+    based on a given pattern and the marker num we wish to generate data for.
+
+    Args:
+        size_img: the dimensions of the image
+        radius: the radius of the cell
+        marker_num: the marker num of the cell we're looking at
+        expression: whether the channel we're looking at is nuclear or membrane
+        pattern: the specific pattern we wish to generate signal by, for now we'll just assume it's equal
+            to expression (nuclear or membrane)
+        cell_region: the region covered by the cell we're processing
+        cell_center: the center of the cell we're processing
+
+    Returns:
+        channel_signal: a NumPy array with the generated signal according to the parameters provided
+    """
+    cell_center_x = cell_center[0]
+    cell_center_y = cell_center[1]
+
+    # create the signal data array
+    signal_data = np.zeros(size_img)
+
+    # create a probability mask, which will become a set of random values based on a normal distribution
+    # either centered at a peak for nuclear, or around the edge for membrane
+    prob_mask = np.zeros(size_img)
+
+    # get the coordinates of every point in the array
+    prob_mask_rows, prob_mask_cols = np.ogrid[:size_img[0], :size_img[1]]
+
+    # generate an array of numbers valued depending on how close or far away the coordinate is
+    # from the center or the membrane
+    # note that we multiply by -random() for the membrane case because the values
+    # will initially be negative and it will help to make the values consistent
+    # also, we need to center the distribution around the center for nuclear
+    # but around an edge value for membrane
+    if expression == "nuclear":
+        prob_mask = size_img[0] + size_img[1] - np.maximum(np.abs(prob_mask_rows - cell_center_x),
```

Comment:
Ah OK. I should include a comment about what is going on here for sure.

Yes, prob_mask is a placeholder for the intermediate calculations. This is because the current way I generate probability values covers the entire 1024x1024 of the image, but I'll only need to extract a certain non-square subset. That being said, I'm going to work on trying not to use a intermediate variable next version.

For nuclear expression, the best way to describe what prob_mask becomes initially is a topographical map of a mountain, with higher values indicating higher elevation. I want the highest value to be at the center and for the values to decrease the further away they get from the center. This is so when I calculate the probability of representation, it returns higher values the closer they are to the center (which I center the distribution around).